### PR TITLE
feat(discover2): Add "save as", "update", and "delete" saved queries into the discover dashboard

### PR DESCRIPTION
**Pull request by dashed**
## TODO

- [x] remove edit link from popout menu
- [x] remove delete link from popout menu
- [x] add `EventView` equality
- [x] add "save as" button
- [x] add "update query" button
- [x] add "delete query" button
- [ ] test: create saved query
- [ ] test: save a saved query as a new query
- [ ] test: delete a saved query
- [ ] test: update a saved query
- [ ] percy tests (may be delegated to a separate PR)

-------

Closes SEN-1030

### CODE REVIEW

File: src/sentry/static/sentry/app/components/sidebar/discover2Item.tsx [hunk 1]
```diff
@@ -67,24 +62,6 @@ class Discover2Item extends React.Component<Props, State> {
...
-    event.stopPropagation();
-    const {organization} = this.props;
-    const target = {
-      pathname: `/organizations/${organization.slug}/eventsv2/`,
-      query: {...EventView.fromSavedQuery(item).generateQueryStringObject(), edit: true},
-    };
-    browserHistory.push(target);
-  };
-
-  handleDelete = (event: React.MouseEvent<Element>, item: SavedQuery) => {
```
Comment (markstory) [inline src/sentry/static/sentry/app/components/sidebar/discover2Item.tsx]: Is this part of the sidebar deprecation plan? I like having the edit/delete closer to the name + save as they are contextually related.

File: src/sentry/static/sentry/app/views/eventsV2/eventView.tsx [hunk 2]
```diff
@@ -421,6 +422,49 @@ class EventView {
     });
   }
 
+  isEqualTo(other: EventView): boolean {
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/eventsV2/eventView.tsx]: Could this also be used in the existing code we have for determining if an API payload differs?

File: src/sentry/static/sentry/app/views/eventsV2/savedQueryButtonGroup.tsx [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+type Props = {
+  api: Client;
+  eventView: EventView;
+  location: Location;
+  organization: Organization;
+  savedQueries: SavedQuery[];
+};
+
+class SavedQueryButtonGroup extends React.Component<Props> {
+  getExistingSavedQuery = (): EventView | undefined => {
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/eventsV2/savedQueryButtonGroup.tsx]: Is it better to use `null` or `undefined` in 'thing or nothing' style methods in typescript?

File: src/sentry/static/sentry/app/views/eventsV2/savedQueryButtonGroup.tsx [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+        {this.renderSaveButton()}
+      </ButtonGroup>
+    );
+  }
+}
+
+const ButtonGroup = styled('div')`
+  display: flex;
+
+  > * + * {
```
Comment (markstory) [inline src/sentry/static/sentry/app/views/eventsV2/savedQueryButtonGroup.tsx]: :owl:

File: src/sentry/static/sentry/app/views/eventsV2/savedQueryButtonGroup.tsx (hunk 3 above)
Comment (dashed) [reply src/sentry/static/sentry/app/views/eventsV2/savedQueryButtonGroup.tsx]: Either `null` or `undefined` works fine. 

We've initially configured TypeScript to treat `null` and `undefined` as two different distinct values.

Comment (markstory) [reply src/sentry/static/sentry/app/views/eventsV2/savedQueryButtonGroup.tsx]: Ok.

File: src/sentry/static/sentry/app/views/eventsV2/savedQueryButtonGroup.tsx (hunk 4 above)
Comment (leedongwei) [reply src/sentry/static/sentry/app/views/eventsV2/savedQueryButtonGroup.tsx]: Lol this is funny. I did something similar too.

Comment (leedongwei) [reply src/sentry/static/sentry/app/views/eventsV2/savedQueryButtonGroup.tsx]:
```
  > * {
    margin: ${space(1)};

    &:last-child {
      margin-right: 0;
    }
  }
```

File: src/sentry/static/sentry/app/views/eventsV2/savedQueryButtonGroup.tsx (hunk 3 above)
Comment (leedongwei) [reply src/sentry/static/sentry/app/views/eventsV2/savedQueryButtonGroup.tsx]: I prefer `undefined` because `typeof null == "object"` is true.

**Review by leedongwei — APPROVED** [review]

**Pull request merged**